marshmallow_jsonschema/base.py

```diff
@@ -268,7 +269,7 @@ def _get_schema_for_field(self, obj, field):
             if isinstance(field, fields.Nested):
                 # Special treatment for nested fields.
                 schema = self._from_nested_schema(obj, field)
-            elif ALLOW_UNIONS and isinstance(field, Union):
+            elif isinstance(field, Union):
```

Comment:
So `isinstance` is nice but it does mean we need to import the appropriate library.
Rather than doing that, we can just look for the property that the field should have that we need to read from.

`_candidate_fields` - https://github.com/adamboche/python-marshmallow-union/blob/465c6d4127d5d2302296ec347c4faabf9b374a65/src/marshmallow_union/__init__.py#L40
`union_fields` - https://github.com/lovasoa/marshmallow_dataclass/blob/master/marshmallow_dataclass/union_field.py#L31

The downside is each of those does return slightly different formats, so you might need to do a loop first to get all the fields appropriately before feeding them into the anyOf list comprehension